zookeeper-server/src/main/java/org/apache/zookeeper/server/admin/ReadAheadEndpoint.java

```diff
@@ -74,7 +74,10 @@ public ReadAheadEndpoint(final EndPoint channel, final int readAheadLength){
 
     private synchronized void readAhead() throws IOException {
         if (leftToRead > 0) {
-            final int n = endPoint.fill(start);
+            int n = 0;
```

Comment:
I discovered in my changes when cleaning things up. I was running into the situation where data a TLS connection wasn't ready to be read (`fill()`). Probably because secure connections have a bigger overhead, `JettyAdminServerTest` would only fail for secure connections (the plaintext connections would work fine). My idea here is to wait until the data is ready to be read, but as I am not an expert with Jetty `EndPoint`, these lines seems to have potential to infinitely loop.